### ISSUE TITLE
WPSWA version 2.0.0 Upgrade Notice

### ISSUE THREAD

**Issue by richaber** [WebDevStudios/wp-search-with-algolia#143]
WP Search with Algolia version 2.0.0 will upgrade the Algolia JavaScript libraries that are bundled with the plugin.

Algolia’s JavaScript libraries have changed significantly since this plugin first shipped.

Upgrading the bundled JS libraries has been a recurring request from both advanced WPSWA users and JavaScript developers alike.

For WPSWA users who have a customized `autocomplete.php` and/or `instantsearch.php` template in their theme, this will be a breaking change, hence the [MAJOR version change](https://semver.org/#summary).

For all other users, those who do not use customized `autocomplete.php` and/or `instantsearch.php` templates, this should be a painless upgrade.

When WPSWA version 2.0.0 ships, we will warn users in WP-Admin if they have customized template files that are lower versions than those that ship with WPSWA version 2.0.0.

Users will be encouraged to remove those outdated template files from their theme.

Advanced users and developers will still be able to copy the templates from the plugin to their theme, replacing the outdated versions, for customization.

We are unable to provide support for customization requests here.

If you require customization beyond what you or your developer can do yourselves, consider [hiring WebDevStudios](https://webdevstudios.com/contact/).

**Comment by gsantos2099** [comment]
Hi @richaber, thanks for the update! Do you have an estimate when the version 2.0.0 will be available? Or is there a testing version I could start playing with? I'm about to launch my project with version 1.7.0, but I maybe I should hold a bit longer... Thank you.

**Comment by mark-c-woodard** [comment]
I saw that 1.8.0 released a few das ago with the 2.0.0 upgrade notices. Does that mean that 2.0.0 will be ready soon?

**Comment by tw2113** [comment]
@mark-c-woodard not going to say that it's going to be imminent, but I am pretty sure we're trying to give ample time for the plugin users to prepare for when we get to it.

**Comment by mark-c-woodard** [comment]
@tw2113 I hear you and thanks for the update.
I just started writing an extension for the plugin and I'm looking to work with some of the newer widgets like rangeInput. Excited for 2.0.0, thanks for keeping this plugin updated!

**Comment by mark-c-woodard** [comment]
I'm currently using the feature/instant-search-update branch for my extension. Is that still kind of the basis for for the 2.0.0 release? 
Just want to make sure I don't get way out of sync with the main repo.

**Comment by richaber** [comment]
> I'm currently using the feature/instant-search-update branch for my extension. Is that still kind of the basis for for the 2.0.0 release?
> Just want to make sure I don't get way out of sync with the main repo.

Yes, that is where we are currently working, and PR #149 will merge that to a 2.0.0 release branch.

**Comment by richaber** [comment]
WP Search with Algolia version 2.0.0 has been tagged and released.

https://github.com/WebDevStudios/wp-search-with-algolia/releases/tag/2.0.0

https://plugins.svn.wordpress.org/wp-search-with-algolia/tags/2.0.0/
